runtime/vm/JFRConstantPoolTypes.hpp

```diff
@@ -413,6 +425,71 @@ class VM_JFRConstantPoolTypes {
 		return;
 	}
 
+	const char *getSystemProp(const char *propName) {
+		J9VMSystemProperty *jvmInfoProperty = NULL;
+		const char *value = "";
+		UDATA getPropertyResult = _vm->internalVMFunctions->getSystemProperty(_vm, propName, &jvmInfoProperty);
+		if (J9SYSPROP_ERROR_NOT_FOUND != getPropertyResult) {
+			value = jvmInfoProperty->value;
+		}
+		return value;
+	}
+
+    void initJVMInfo() {
+		_jvmInformation = {0};
+
+		/* Set JVM name */
+		_jvmInformation.jvmName = getSystemProp("java.vm.name");
+
+		/* Set JVM version */
+		_jvmInformation.jvmVersion = getSystemProp("java.vm.info");
+
+		/* Set Java arguments */
+		JavaVMInitArgs* vmArgs = _vm->vmArgsArray->actualVMArgs;
+		const char *javaCommand = "-Dsun.java.command=";
+		UDATA javaArgsLen = 0;
+		for (UDATA i = 0; i < vmArgs->nOptions; i++) {
+			if (0 == strncmp(vmArgs->options[i].optionString, javaCommand, strlen(javaCommand))) {
+				javaArgsLen = strlen(vmArgs->options[i].optionString) - strlen(javaCommand);
+				_jvmInformation.javaArguments = vmArgs->options[i].optionString + strlen(javaCommand);
+				break;
+			}
+		}
+
+		/* Get JVM arguments from command line by removing the command name and Java arguments */
+		const char *envVarName = "OPENJ9_JAVA_COMMAND_LINE";
```

Comment:
The command line may not contain all the JVM args, youll need to walk the `vm->vmArgsArray->actualVMArgs` to get the length of all optinos (that start with `-`) then allocate the array, then walk the list again to write all the args.

Reply:
Looks like all the options in `actualVMArgs` start with `-`:
```
-Xoptionsfile=/Users/gengchentuo/openj9-openjdk-jdk21/build/macosx-aarch64-server-release/images/jdk/lib/options.default
-Xlockword:mode=default,noLockword=java/lang/String,noLockword=java/util/MapEntry,noLockword=java/util/HashMap$Entry,noLockword=org/apache/harmony/luni/util/ModifiedMap$Entry,noLockword=java/util/Hashtable$Entry,noLockword=java/lang/invoke/MethodType,noLockword=java/lang/invoke/MethodHandle,noLockword=java/lang/invoke/CollectHandle,noLockword=java/lang/invoke/ConstructorHandle,noLockword=java/lang/invoke/ConvertHandle,noLockword=java/lang/invoke/ArgumentConversionHandle,noLockword=java/lang/invoke/AsTypeHandle,noLockword=java/lang/invoke/ExplicitCastHandle,noLockword=java/lang/invoke/FilterReturnHandle,noLockword=java/lang/invoke/DirectHandle,noLockword=java/lang/invoke/ReceiverBoundHandle,noLockword=java/lang/invoke/DynamicInvokerHandle,noLockword=java/lang/invoke/FieldHandle,noLockword=java/lang/invoke/FieldGetterHandle,noLockword=java/lang/invoke/FieldSetterHandle,noLockword=java/lang/invoke/StaticFieldGetterHandle,noLockword=java/lang/invoke/StaticFieldSetterHandle,noLockword=java/lang/invoke/IndirectHandle,noLockword=java/lang/invoke/InterfaceHandle,noLockword=java/lang/invoke/VirtualHandle,noLockword=java/lang/invoke/PrimitiveHandle,noLockword=java/lang/invoke/InvokeExactHandle,noLockword=java/lang/invoke/InvokeGenericHandle,noLockword=java/lang/invoke/VarargsCollectorHandle,noLockword=java/lang/invoke/ThunkTuple
-XX:+EnsureHashed:java/lang/Class,java/lang/Thread
-Xjcl:jclse29
-Dcom.ibm.oti.vm.bootstrap.library.path=/Users/gengchentuo/openj9-openjdk-jdk21/build/macosx-aarch64-server-release/images/jdk/lib/default:/Users/gengchentuo/openj9-openjdk-jdk21/build/macosx-aarch64-server-release/images/jdk/lib
-Dsun.boot.library.path=/Users/gengchentuo/openj9-openjdk-jdk21/build/macosx-aarch64-server-release/images/jdk/lib/default:/Users/gengchentuo/openj9-openjdk-jdk21/build/macosx-aarch64-server-release/images/jdk/lib
-Djava.library.path=/Users/gengchentuo/openj9-openjdk-jdk21/build/macosx-aarch64-server-release/images/jdk/lib/default:/Users/gengchentuo/openj9-openjdk-jdk21/build/macosx-aarch64-server-release/images/jdk/lib:/usr/lib
-Djava.home=/Users/gengchentuo/openj9-openjdk-jdk21/build/macosx-aarch64-server-release/images/jdk
-Duser.dir=/Users/gengchentuo/IdeaProjects/playground/out/production/playground
-XX:+FlightRecorder
--add-exports=java.base/com.ibm.oti.vm=ALL-UNNAMED
-Djava.class.path=.
-Dsun.java.command=JFRTest arggg fafda
-Dsun.java.launcher=SUN_STANDARD
```
do we want all of them?

Reply:
yes, lets keep everything